suseconnect/migration.go

```diff
@@ -457,7 +367,7 @@ func isSnapperConfigured() bool {
 	return false
 }
 
-func checkSystemProducts() ([]connect.Product, error) {
+func checkSystemProducts(rollbackOnFailure bool, printInstalledProducts bool) ([]connect.Product, error) {
```

Comment:
Yeah, I got to the same idea when I looked at this code after I finished. There are currently only two cases where this code is used and they use mutually exclusive set of options so it might as well be restructured to have these parts outside.

Reply:
I moved the printing part out.
I left the rollback part in as I think it is kind of part of the product checking functionality (even if optional).